### ISSUE TITLE
AndroidとiOS両方のインストール用リンクを共有する

### ISSUE THREAD

**Issue by yusuke** [cocoa-mhlw/cocoa#251]
## その機能リクエストは何らかの問題に関連しますか / Is your feature request related to a problem?

現在iOSで共有するとiTunes Storeのリンクのみが共有されます。Androidユーザーはそのリンクを辿ってもインストールできません。

## 解決策についてお書きください / Describe the solution you'd like
iOSでもAndroidでも、Google Play/iTunes Store両方のリンクを共有する、または両方のリンクが含まれるランディングページを共有する。

iOSユーザーが共有したものを見た歳、Androidユーザーも手軽にインストールできるようにする。逆も然り。

**Comment by b-wind** [comment]
共有用のランディングページ見たいなのは有った方が良いと思います。
ただ、その辺りのお話はGitHub上ではしないことになっているような気がします。

厚生労働省に要望を送るのが一応正規のルートだと認識していますね。

**Comment by Takym** [comment]
> ただ、その辺りのお話はGitHub上では扱わないことになっているような気がします。（微妙なラインですが）

共有ボタンの挙動を変更するという点では、GitHub 上でも扱う事はできる気がします。

**Comment by b-wind** [comment]
> 共有ボタンの挙動を変更するという点では、GitHub 上でも扱う事はできる気がします。

なるほど。GitHub 上で扱える範囲で改善すると言う案なら実現出来るかもですね。

**Comment by yusuke** [comment]
ランディングページを作りましょうという提案ではありません。

スタート地点は「iOSユーザのシェアしたリンクからAndroidユーザもインストールできるように」「AndridユーザのシェアしたリンクからiOSユーザもインストールできるように」するという点にあります。

実装方法としては

- iTunes Store / Google Play Store両方のリンクをシェアする仕組みにする
- OSを判別して自動的にiTunes Store / Google Play Storeへリダイレクトするページをシェアする仕組みにする(og,twitterタグもあると綺麗にシェアできていいですね)
- ランディングページを作ってそのページをシェアする仕組みにする

などが考えられます。

**Comment by yusuke** [comment]
ランディングページを作るのはアプリに閉じた話ではないからアプリだけで実現する方式に絞って検討する、という必要もないと思います。

**Comment by b-wind** [comment]
補足ですが、 Android 版では PlayStore へのリンクを共有するようになっていますね。
少なくともOS毎に共有内容を分ける必然性は無さそうです。

**Comment by b-wind** [comment]
この Issue とは直接関係ないのですが、共有リンクについての扱いがこちら（https://github.com/cocoa-mhlw/cocoa/issues/194#issuecomment-851897713）でコメントされています。

ご参考までに。

**Comment by yusuke** [comment]
このページをシェアするのも一案ですね。日本語のページのみのようですが。
https://www.mhlw.go.jp/stf/seisakunitsuite/bunya/cocoa_00138.html

**Comment by b-wind** [comment]
> https://www.mhlw.go.jp/stf/seisakunitsuite/bunya/cocoa_00138.html


今回初めてチェックしましたが一応モバイルでも見れるようになっているんですね（いつもPCで見ていた）
ランディングページとしては少々ごちゃごちゃしている気もしますが、新しく作らなくて良いのは都合が良いです。
mhlw.go.jp ドメイン名配下なのもベターに思えます。

**Comment by keiji** [comment]
両方のプラットフォーム向けのリンクを併記した方が良いのは確かにそのとおりだと思います。

現状、なぜAndroidとiOSでアプリのリンクを別にしているのか（技術的にはそちらの方が面倒さそうなんですが）、まずは確認してみたいと思います。

**Comment by yusuke** [comment]
シェアはここでやっているみたいですね。
https://github.com/cocoa-mhlw/cocoa/blob/develop/Covid19Radar/Covid19Radar/Common/AppUtils.cs#L27

**Comment by keiji** [comment]
ありがとうございます。
履歴を見る限りCovid19Radarの頃からこの仕様だったようで、一度ぼくの手元で共通化してリンクを2つ掲載できるか試してみます。できるとなれば、あらためて進めていきましょう。

**Comment by yusuke** [comment]
はい。単に https://www.mhlw.go.jp/stf/seisakunitsuite/bunya/cocoa_00138.html をシェアする仕組みでもいいんじゃないかなという気もしますが。
(og:imageタグが空なのは直すとして)

**Comment by yusuke** [comment]
#250 で触れましたが、シェアする際にインストールを促進するような文面が記載されているといいなと思います。
「COCOAをインストールして命を救おう！」
「COCOAを自分を、大切な人を守ろう！」
「万が一の際も接触通知を受け取れば安心」
みたいな(すいません、決して上のキャッチが良いとは自分でも思っていませんが)。

**Comment by keiji** [comment]
昨日、喫茶店で見たCOCOAのQRコードは  https://www.mhlw.go.jp/stf/seisakunitsuite/bunya/cocoa_00138.html がリンク先でした。

![PXL_20210704_070334445](https://user-images.githubusercontent.com/932136/124420770-034d5080-dd9b-11eb-9bfc-1a43ed60bc17.jpg)
